.github/dependabot.yml

```diff
@@ -1,7 +1,13 @@
-# Check for updates to GitHub Actions
 version: 2
 updates:
   - package-ecosystem: "github-actions"
     directory: "/"
+    labels: ["dependencies"]
     schedule:
-      interval: "daily"
+      interval: "weekly"
+  - package-ecosystem: "pip"
+    directory: "/"
+    labels: ["dependencies"]
+    versioning-strategy: increase
+    schedule:
+      interval: "monthly"
```

Comment:
We were missing this section to update our pip dependencies. Thus, `ruff`, `mypy` etc. are actually quite out-of-date.
I will update those in a follow-up PR.